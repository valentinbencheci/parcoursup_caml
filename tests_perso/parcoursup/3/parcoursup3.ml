(* ###################### - DESCRIPTION TEST - ###################### *)
(* Ce test est très intéressant, car lorsqu'Adam abandonne, les listes 
de 2 personnes sont régénérées. *)
(* ################################################################## *)

open Test_parcoursup_utils

let session = nouvelle_session ()

let () = 
  ajoute_candidat session ~nom_candidat:"Adam";
  ajoute_candidat session ~nom_candidat:"Olivier";
  ajoute_candidat session ~nom_candidat:"Marie";
  ajoute_candidat session ~nom_candidat:"Pierre";
  ajoute_formation session ~nom_formation:"Université Côte d'Azur" ~capacite:2;
  ajoute_formation session ~nom_formation:"Université de Toulon" ~capacite:2;
  ajoute_formation session ~nom_formation:"Université de Paris" ~capacite:1;
  ajoute_voeu session 
    ~rang_repondeur:(Some 1) 
    ~nom_candidat:"Adam" 
    ~nom_formation:"Université Côte d'Azur";
  ajoute_voeu session 
    ~rang_repondeur:(None) 
    ~nom_candidat:"Adam" 
    ~nom_formation:"Université de Paris";
  ajoute_voeu session 
    ~rang_repondeur:(Some 2) 
    ~nom_candidat:"Olivier" 
    ~nom_formation:"Université de Paris";
  ajoute_voeu session 
    ~rang_repondeur:(None) 
    ~nom_candidat:"Olivier" 
    ~nom_formation:"Université Côte d'Azur";
  ajoute_voeu session 
    ~rang_repondeur:(Some 4) 
    ~nom_candidat:"Olivier" 
    ~nom_formation:"Université de Toulon";
  ajoute_voeu session 
    ~rang_repondeur:(Some 5) 
    ~nom_candidat:"Marie" 
    ~nom_formation:"Université de Toulon";
  ajoute_voeu session 
    ~rang_repondeur:(Some 5) 
    ~nom_candidat:"Pierre" 
    ~nom_formation:"Université de Toulon";

  ajoute_commission session 
    ~nom_formation:"Université Côte d'Azur" 
    ["Adam";"Olivier";"Marie";"Pierre"]; 
  ajoute_commission session 
    ~nom_formation:"Université de Toulon" 
    ["Adam";"Olivier";"Marie";"Pierre"];  
  ajoute_commission session 
    ~nom_formation:"Université de Paris" 
    ["Adam";"Olivier";"Marie";"Pierre"]; 
    
  reunit_commissions session;

  nouveau_jour session;
  affiche_voeux_en_attente session "Adam";
  affiche_propositions_en_attente session "Adam";
  affiche_voeux_en_attente session "Olivier";
  affiche_propositions_en_attente session "Olivier";
  affiche_voeux_en_attente session "Marie";
  affiche_propositions_en_attente session "Marie";
  affiche_voeux_en_attente session "Pierre";
  affiche_propositions_en_attente session "Pierre";

  renonce session 
    ~nom_candidat:"Adam" 
    ~nom_formation:"Université de Paris";

  nouveau_jour session;
  affiche_voeux_en_attente session "Adam";
  affiche_propositions_en_attente session "Adam";
  affiche_voeux_en_attente session "Olivier";
  affiche_propositions_en_attente session "Olivier";
  affiche_voeux_en_attente session "Marie";
  affiche_propositions_en_attente session "Marie";
  affiche_voeux_en_attente session "Pierre";
  affiche_propositions_en_attente session "Pierre";