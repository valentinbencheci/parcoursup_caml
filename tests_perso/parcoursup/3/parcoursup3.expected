nouvelle session
ajoute candidat Adam
ajoute candidat Olivier
ajoute candidat Marie
ajoute candidat Pierre
ajoute formation Université Côte d'Azur (2 places)
ajoute formation Université de Toulon (2 places)
ajoute formation Université de Paris (1 places)
ajoute voeu de Adam pour Université Côte d'Azur (rang=1)
ajoute voeu de Adam pour Université de Paris
ajoute voeu de Olivier pour Université de Paris (rang=2)
ajoute voeu de Olivier pour Université Côte d'Azur
ajoute voeu de Olivier pour Université de Toulon (rang=4)
ajoute voeu de Marie pour Université de Toulon (rang=5)
ajoute voeu de Pierre pour Université de Toulon (rang=5)
commission de classement de Université Côte d'Azur : [Adam; Olivier; Marie; Pierre]
commission de classement de Université de Toulon : [Adam; Olivier; Marie; Pierre]
commission de classement de Université de Paris : [Adam; Olivier; Marie; Pierre]
réunit commissions
nouveau jour
Voeux en attente de Adam : {}
Propositions en attente de Adam : {Université Côte d'Azur, Université de Paris}
Voeux en attente de Olivier : {Université de Paris(0)}
Propositions en attente de Olivier : {Université Côte d'Azur, Université de Toulon}
Voeux en attente de Marie : {}
Propositions en attente de Marie : {Université de Toulon}
Voeux en attente de Pierre : {Université de Toulon(0)}
Propositions en attente de Pierre : {}
Adam renonce à Université de Paris
nouveau jour
Voeux en attente de Adam : {}
Propositions en attente de Adam : {Université Côte d'Azur}
Voeux en attente de Olivier : {}
Propositions en attente de Olivier : {Université Côte d'Azur, Université de Paris}
Voeux en attente de Marie : {}
Propositions en attente de Marie : {Université de Toulon}
Voeux en attente de Pierre : {}
Propositions en attente de Pierre : {Université de Toulon}
